utilities/semaphore.py

```diff
@@ -0,0 +1,47 @@
+import time
+import uuid
+
+from threading import Thread, BoundedSemaphore
+
+
+class ThreadManager(object):
+    def __init__(self, semaphore_count: int | None = None, delay: float = 0.25):
+        if semaphore_count is None:
+            semaphore_count = 8
```

Comment:
This actually can never happen. We're passing the 12 threads from the default click value.
```suggestion

```